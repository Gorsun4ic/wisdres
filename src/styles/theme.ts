interface Theme {
	colors: Record<string, string>; // More specific type for colors
	fontWeights?: Record<string, string>; // More specific type for fontWeights
	borders?: Record<string, string>; // More specific type for borders
	fontSizes?: Record<string, string>;
}

const theme: Theme = {
	colors: {
		black: "#000",
		grey: "#7A7A7A",
		darkGrey: "#D9D9D9",
		formBackground: "#F6F6F6",
		yellow: "#FFCD36",
		white: "#fff",
		green: "#5EB168",
		darkGreen: "#519a5a",
	},
	fontWeights: {
		bold: "700",
		medium: "500",
		regular: "400",
	},
	fontSizes: {
		default: "1rem",
		h1: "2.25rem",
		h2: "1.75rem",
		h3: "1.25rem",
		h4: "1.125rem",
	},
	borders: {
		borderRadius: "2px",
		mainBorder: "1px",
		lightBorder: "0.5px",
	},
};

export default theme;
