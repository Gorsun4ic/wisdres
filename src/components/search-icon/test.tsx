// Import React and necessary libraries
import { render } from "@testing-library/react";
import theme from "@styles/theme";
import { ThemeProvider } from "styled-components";
import SearchIcon from "./index"; // Assuming the component is called SearchIcon
import "@testing-library/jest-dom"; // Importing Jest DOM for additional matchers

describe("SearchIcon Component", () => {
	// Test case for default render
	it("renders correctly with default props", () => {
		const { container } = render(<SearchIcon />);
		const svg = container.querySelector("svg");
		const path = container.querySelector("path");

		// Check if SVG is rendered
		expect(svg).toBeInTheDocument();
		expect(svg).toHaveAttribute("width", "24");
		expect(svg).toHaveAttribute("height", "24");
		expect(path).toHaveAttribute("stroke", theme.colors.grey);
	});

	// Test case for custom size prop
	it("renders with custom size", () => {
		const customSize = 50;
		const customColor = "red";
		const { container } = render(
			<ThemeProvider theme={theme}>
				<SearchIcon size={customSize} color={customColor} />
			</ThemeProvider>
		);
		const svg = container.querySelector("svg");
		const path = container.querySelector("path");

		// Check if the size is applied
		expect(svg).toHaveAttribute("width", customSize.toString());
		expect(svg).toHaveAttribute("height", customSize.toString());
		expect(path).toHaveAttribute("stroke", customColor);
	});
});
