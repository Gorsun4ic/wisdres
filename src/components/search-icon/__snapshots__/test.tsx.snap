// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchIcon Component > matches the snapshot 1`] = `
<svg
  fill="none"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M16.9268 17.04L20.4 20.4M19.28 11.44C19.28 15.7699 15.7699 19.28 11.44 19.28C7.11006 19.28 3.59998 15.7699 3.59998 11.44C3.59998 7.11009 7.11006 3.60001 11.44 3.60001C15.7699 3.60001 19.28 7.11009 19.28 11.44Z"
    stroke="#7A7A7A"
    stroke-linecap="round"
  />
</svg>
`;
